chanotify/chanotify_test.go

```diff
@@ -78,3 +78,16 @@ func TestConcurrentNotifier(t *testing.T) {
 		}
 	}
 }
+
+func TestAddToBlocked(t *testing.T) {
+	s := New()
+	ch := make(chan struct{}, 1)
+	// give some time to start first select
```

Comment:
maybe this test is better as:
```
go func() {
  time.Sleep(1 * time.Second)
  s.Add(ch, "1")
  ch <- struct{}{}
}()
val := <-s.Chan()
```

Atm, if you don't know the internals you would not know if this was because adding while being blocked is supported or because only first call to `s.Chan()` generates the condition.